src/pages/ImportData.js

```diff
@@ -77,40 +77,224 @@ const ResponseFragment = ({ title, snapshotResponses, responseIndex }) => {
+        }, workspaces),
+        value: selectedWorkspaceId,
+        onChange: setSelectedWorkspaceId
+      })
+    ])]),
+    isDataset && div({ style: { marginTop: '0.5rem', lineHeight: '1.5' } }, [noteMessage]),
+    div({ style: { display: 'flex', alignItems: 'center', marginTop: '1rem' } }, [
+      h(ButtonSecondary, { onClick: setMode, style: { marginLeft: 'auto' } }, ['Back']),
+      h(ButtonPrimary, {
+        style: { marginLeft: '2rem' },
+        disabled: !selectedWorkspace,
+        onClick: () => onImport(selectedWorkspace.workspace)
+      }, ['Import'])
+    ])
+  ])
+
+  const SelectTemplateWorkspace = () => h(Fragment, [
+    h2({ style: styles.title }, ['Start with a template']),
+    isDataset && div({ style: { marginBottom: '1rem', lineHeight: '1.5' } }, [noteMessage]),
+    div({
+      role: 'radiogroup',
+      'aria-label': 'choose a template',
+      style: { overflow: 'auto', maxHeight: '25rem' }
+    }, [
+      _.map(([i, ws]) => {
+        const { name, namespace, description, hasNotebooks, hasWorkflows } = ws
+        const isSelected = _.isEqual({ name, namespace }, selectedTemplateWorkspaceKey)
+
+        return div({
+          key: `${name}/${namespace}`,
+          style: {
+            display: 'flex', alignItems: 'baseline',
+            marginBottom: '1rem', paddingLeft: '0.25rem',
+            ...(i > 0 ? { borderTop: Style.standardLine, paddingTop: '1rem' } : {})
+          }
+        }, [
+          h(RadioButton, {
+            name: 'select-template',
+            checked: isSelected,
+            onChange: () => setSelectedTemplateWorkspaceKey({ namespace, name }),
+            text: h(Collapse, {
+              style: { fontSize: 14, marginLeft: '0.5rem' },
+              title: span({ style: { display: 'flex', alignItems: 'center' } }, [
+                span({ style: { fontWeight: 600 } }, [name]),
+                hasNotebooks && img({ src: jupyterLogo, style: { height: 23, width: 23, marginLeft: '0.5rem' } }),
+                hasWorkflows &&
+                  wdlIcon({ style: { height: 23, width: 23, marginLeft: '0.5rem', borderRadius: 3, padding: '8px 4px 7px 4px' } })
+              ])
+            }, [p({ style: { fontSize: 14, lineHeight: '1.5', marginRight: '1rem' } }, [description])])
+          })
+        ]
+        )
+      }, Utils.toIndexPairs(filteredTemplates))
+    ]),
+    div({ style: { display: 'flex', alignItems: 'center', marginTop: '1rem' } }, [
+      h(ButtonSecondary, { style: { marginLeft: 'auto' }, onClick: setMode }, ['Back']),
+      h(ButtonPrimary, {
+        style: { marginLeft: '2rem' },
+        disabled: !selectedTemplateWorkspaceKey,
+        onClick: () => setIsCloneOpen(true)
+      }, ['Import'])
+    ])
+  ])
+
+  return div({ style: { ...styles.card, marginLeft: '2rem' } }, [
+    Utils.switchCase(mode,
+      ['existing', () => h(SelectExistingWorkspace)],
+      ['template', () => h(SelectTemplateWorkspace)],
+      [Utils.DEFAULT, () => {
```

Comment:
I left Utils.DEFAULT here because its hard to name well, and it doesn't hurt understanding of what options are in the switch to leave as this.